Apps/W1/HybridBaseDeployment/app/src/codeunits/CloudMigReplicateDataMgt.Codeunit.al

```diff
@@ -71,6 +71,46 @@ codeunit 40021 "Cloud Mig. Replicate Data Mgt."
         exit(CanBeIncluded);
     end;
 
+    procedure CheckAndIncludeExcludeTableListFromCloudMigration(TableIDList: List of [Integer]; NewReplicateData: Boolean)
```

Comment:
I do not like this method. There is no strong reason to tie ourselves to a public interface that forces everyone to provide a list of integers. It can also cause a lot of issues when having thousands of records to update. 

Can we get a method that enables a single table, then you can decide if you want to write like this
IncludeTableToReplication(Database::"Change Log");
IncludeTableToReplication(Database::"Change Log Entry");
IncludeTableToReplication(Database::"My Change Log");
ExcludeTableFromReplciation(Database::"Customer);

We need 2 method types for replication:
IncludeTableToReplication(TableID)
ExcludeTableFromReplication(TableID)

Alternatevelly we need the following methods too, to specify how the delta sync should work:
SetReplaceCloudData(TableID)
SetPreserveCloudData(TableID) 

IncludeTableToReplication/ExcludeTableFromReplication could get an overload with preserve cloud data so you can set it in one go.

Reply:
I would actually prefer a function for a single table only. In this case you would probably get hundreds of telemetry entries. If you prefer that path anyway, I will change or better said simplify the function. My goal was to be aligned with how the current function works in the context of sending telemetry data.
![image](https://github.com/user-attachments/assets/f99e12e7-6658-4cff-b346-c8eca39a1925)

Replace and Preserve - agreed. I can do that.

Reply:
The problem with the list is that there are multiple properties that control how the table is replicated.

Currently there is:
Table ID and Company name identify a table
Replicate, Replace or delta sync control how the data is replicated.

There could be more in the future, I'm not sure how the table extensions are handled, so we may need an extra property. 

For these reasons a list of integers is not ideal. I would recommend that we implement dedicated methods to update a single table first, then we can always add overloads if needed.

Reply:
Sounds even better as far as I am concerned.

Reply:
"Per table" functions added.